[BUG] repository-multi-version bwc test is failing for elasticsearch version V7.10.2

**Describe the bug**

The backwards compatibility tests for Elasticsearch version v7.10.2 are failing for the `repository-multi-version`. These tests verify if a snapshot repository is not getting corrupted during the cluster upgrade.

Failing Tests:

`testCreateAndRestoreSnapshot`
`testReadOnlyRepo`

**To Reproduce**
`./gradlew :qa:repository-multi-version:v7.10.2#Step3OldClusterTest`

**Expected behavior**
The tests should pass

**Host/Environment (please complete the following information):**
 - OS: Amazon Linux 2
 - Version 2

**Additional context**
Add any other context about the problem here.

Debugging details:

When the downgrade happens from the new cluster to the old cluster in step3, the master dies abruptly with the following error and the other node is not able to find it:

```
[2021-10-20T18:28:42,191][ERROR][o.e.b.ElasticsearchUncaughtExceptionHandler] [v7.10.2-old-0] fatal error in thread [elasticsearch[v7.10.2-old-0][snapshot][T#5]], exiting
java.lang.AssertionError: Index meta generations should have been empty but was [IndexMetaDataGenerations{lookup:{snapshot-step1/9AP-cr3sSr2ofoTwPxkoZQ={[test-index/QG1E38c1ToO69nqfpCYLlA]=F5ZuvDXHQp6tRe8OkrTIDw-_na_-1-1-1}, snapshot-step2/xsVQY29xRvCdmFQQtsb7Gg={[test-index/QG1E38c1ToO69nqfpCYLlA]=0jPix1rNQN-cgFwAEpe63g-_na_-1-1-1}}}{identifier:{0jPix1rNQN-cgFwAEpe63g-_na_-1-1-1=Crf2nnwBEBlHYuDOY2JI, F5ZuvDXHQp6tRe8OkrTIDw-_na_-1-1-1=IJP2nnwBUvrhpEpvOKjy}}]
        at org.elasticsearch.repositories.RepositoryData.addSnapshot(RepositoryData.java:293) ~[elasticsearch-7.10.2.jar:7.10.2]
        at org.elasticsearch.repositories.blobstore.BlobStoreRepository.lambda$finalizeSnapshot$38(BlobStoreRepository.java:1110) ~[elasticsearch-7.10.2.jar:7.10.2]
        at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:63) ~[elasticsearch-7.10.2.jar:7.10.2]
        at org.elasticsearch.action.support.GroupedActionListener.onResponse(GroupedActionListener.java:66) ~[elasticsearch-7.10.2.jar:7.10.2]
        at org.elasticsearch.action.ActionRunnable$1.doRun(ActionRunnable.java:46) ~[elasticsearch-7.10.2.jar:7.10.2]
        at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:743) ~[elasticsearch-7.10.2.jar:7.10.2]
        at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:37) ~[elasticsearch-7.10.2.jar:7.10.2]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) ~[?:?]
        at java.lang.Thread.run(Thread.java:832) [?:?]
```

This happens while [creating a snapshot](https://github.com/opensearch-project/OpenSearch/blob/main/qa/repository-multi-version/src/test/java/org/opensearch/upgrades/MultiVersionRepositoryAccessIT.java#L131) for step3. This failure happens in the old cluster at [RepositoryData](https://github.com/elastic/elasticsearch/blob/7.10/server/src/main/java/org/elasticsearch/repositories/RepositoryData.java#L293).

The index metas are empty for this step even after the index is created successfully for the step3 which reaches [this condition](https://github.com/elastic/elasticsearch/blob/7.10/server/src/main/java/org/elasticsearch/repositories/RepositoryData.java#L291).

I was able to print the details for step 2 since that is in the current version of OpenSearch (step 3 is in the old cluster so I was not able to get the details for it). For step 2, the index metas are not empty:

```
index: {test-index=[test-index/QG1E38c1ToO69nqfpCYLlA]}
Indices GET
health status index      uuid                   pri rep docs.count docs.deleted store.size pri.store.size
green  open   test-index 0jPix1rNQN-cgFwAEpe63g   3   0          0            0       416b           416b
indexmetablobs: {[test-index/QG1E38c1ToO69nqfpCYLlA]=0jPix1rNQN-cgFwAEpe63g-_na_-1-1-1}
```

--- a/qa/repository-multi-version/build.gradle
+++ b/qa/repository-multi-version/build.gradle
@@ -76,16 +76,28 @@ for (Version bwcVersion : BuildParams.bwcVersions.indexCompatible) {
     systemProperty 'tests.rest.suite', 'step2'
   }
 
-  tasks.register("${baseName}#Step3OldClusterTest", StandaloneRestIntegTestTask) {
+  // Step 3 and Step 4 registered for versions for OpenSearch 
+  // since the ES cluster would not be able to read snapshots from OpenSearch cluster in Step 3.
+  if (bwcVersion.after('7.10.2')) {
+    tasks.register("${baseName}#Step3OldClusterTest", StandaloneRestIntegTestTask) {
     useCluster testClusters."${oldClusterName}"
     dependsOn "${baseName}#Step2NewClusterTest"
     systemProperty 'tests.rest.suite', 'step3'
   }
 
-  tasks.register("${baseName}#Step4NewClusterTest", StandaloneRestIntegTestTask) {
-    useCluster testClusters."${newClusterName}"
-    dependsOn "${baseName}#Step3OldClusterTest"
-    systemProperty 'tests.rest.suite', 'step4'
+    tasks.register("${baseName}#Step4NewClusterTest", StandaloneRestIntegTestTask) {
+      useCluster testClusters."${newClusterName}"
+      dependsOn "${baseName}#Step3OldClusterTest"
+      systemProperty 'tests.rest.suite', 'step4'
+    }
+
+    tasks.register(bwcTaskName(bwcVersion)) {
+      dependsOn tasks.named("${baseName}#Step4NewClusterTest")
+    }
+  } else {
+    tasks.register(bwcTaskName(bwcVersion)) {
+      dependsOn tasks.named("${baseName}#Step2NewClusterTest")
+    }
   }
 
   tasks.matching { it.name.startsWith(baseName) && it.name.endsWith("ClusterTest") }.configureEach {
@@ -94,10 +106,6 @@ for (Version bwcVersion : BuildParams.bwcVersions.indexCompatible) {
     def clusterName = it.name.contains("Step2") || it.name.contains("Step4") ? "${newClusterName}" : "${oldClusterName}"
     it.nonInputProperties.systemProperty('tests.rest.cluster', "${-> testClusters."${clusterName}".allHttpSocketURI.join(",")}")
     it.nonInputProperties.systemProperty('tests.clustername', "${-> testClusters."${clusterName}".getName()}")
-  }
-
-  tasks.register(bwcTaskName(bwcVersion)) {
-    dependsOn tasks.named("${baseName}#Step4NewClusterTest")
   }
 }
 

